radix-engine/src/engine/modules/auth_module.rs

```diff
@@ -282,13 +231,66 @@ impl AuthModule {
         Ok(())
     }
 
+    pub fn on_call_frame_enter<Y: SystemApi>(
+        call_frame_update: &mut CallFrameUpdate,
+        actor: &ResolvedActor,
+        system_api: &mut Y,
+    ) -> Result<(), RuntimeError> {
+        let refed = system_api.get_visible_node_ids()?;
+        let auth_zone_id = refed
+            .into_iter()
+            .find(|e| matches!(e, RENodeId::AuthZoneStack(..)))
+            .unwrap();
+        call_frame_update.node_refs_to_copy.insert(auth_zone_id);
+
+        if !matches!(
+            actor.identifier,
+            FnIdentifier::Native(NativeFn::AuthZoneStack(..))
+        ) {
```

Comment:
For my understanding, why not auth zone stack actor?

Reply:
This is self-reference problem. From what I recall, it's easiest to manage self-reference at this layer e.g. so that we can directly reference the last AuthZone in the stack when needing to access the caller's AuthZone rather than always needing to second to last AuthZone.